HLTrigger/HLTfilters/plugins/HLTP2GTTauFilter.h

```diff
@@ -0,0 +1,23 @@
+#ifndef HLTP2GTTauFilter_h
+#define HLTP2GTTauFilter_h
+
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "DataFormats/L1Trigger/interface/P2GTCandidate.h"
+#include "DataFormats/L1Trigger/interface/P2GTAlgoBlock.h"
+
+class HLTP2GTTauFilter : public HLTFilter {
+public:
+  explicit HLTP2GTTauFilter(const edm::ParameterSet&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  bool hltFilter(edm::Event&, const edm::EventSetup&, trigger::TriggerFilterObjectWithRefs&) const override;
+
+private:
+  edm::InputTag m_l1GTAlgoBlockTag;
+  edm::EDGetTokenT<l1t::P2GTAlgoBlockMap> m_algoBlockToken;
+  std::vector<std::string> m_l1GTAlgoNames;
+  double m_minPt;
+  unsigned int m_minN;
+  double m_maxAbsEta;
+};
```

Comment:
any reason to have a separate header file from the plugin implementation? (normally it should not the case for plugins, though some old code still does that).